src/main/java/graphql/schema/GraphQLObjectType.java

```diff
@@ -82,9 +82,9 @@ void replaceInterfaces(List<GraphQLNamedOutputType> interfaces) {
         this.replacedInterfaces = ImmutableList.copyOf(sortTypes(interfaceComparator, interfaces));
     }
 
-    private ImmutableMap<String, GraphQLFieldDefinition> buildDefinitionMap(List<GraphQLFieldDefinition> fieldDefinitions) {
-        return ImmutableMap.copyOf(FpKit.getByName(fieldDefinitions, GraphQLFieldDefinition::getName,
-                (fld1, fld2) -> assertShouldNeverHappen("Duplicated definition for field '%s' in type '%s'", fld1.getName(), this.name)));
+    private Map<String, GraphQLFieldDefinition> buildDefinitionMap(List<GraphQLFieldDefinition> fieldDefinitions) {
+        return FpKit.getByName(fieldDefinitions, GraphQLFieldDefinition::getName,
+                (fld1, fld2) -> assertShouldNeverHappen("Duplicated definition for field '%s' in type '%s'", fld1.getName(), this.name));
```

Comment:
yup - just checked - this does not change the immutability of what we give out since we never give out the map of fields directly

Reply:
That said building object types is done once per schema - so this really wont affect runtime query performance at all I think - or is ImmutableMap lookup of fields somehow slower ???

Reply:
Yeah, `HashMap` and therefore `LinkedHashMap` compares hashes and identity equality first, so is particularly suited for `String` keys.

Reply:
Oh, I should add, this matters less than when I started working on this, because I found I could reduce the calls from 3 to 1, but it'll still help.